tests/system/test_system_dbapi.py

```diff
@@ -426,6 +427,28 @@ def test_DDL_commit(self):
         cur.execute("DROP TABLE Singers")
         conn.commit()
 
+    def test_read_only(self):
```

Comment:
It would be good to add an extra system test that shows the following:
1. A read-only transaction can read data that was present in the database when the read-only transaction started.
2. If data is added using a separate read/write transaction (i.e. another user added data to the database), those records are __not__ visible to the read-only transaction in step 1.
3. __After__ committing/rolling back (closing) the read-only transaction, and starting a new read-only transaction, the data from step 2 should be visible to the read-only transaction.

Reply:
Done